### PR TITLE
Backend event database

### PR DESCRIPTION
**Pull request by fileames**
Event model and validators are created. SportsEvent schema naming conventions are used which can be found [here](https://schema.org/SportsEvent). These models and validators can be used on event related endpoints.

### CODE REVIEW

File: back-end/sports_platform/sports_platform_api/models/event_models.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    description = models.TextField(blank=True)
+
+    startDate = models.DateTimeField()
+
+    latitude = models.DecimalField(max_digits=9, decimal_places=6)
+    longitude = models.DecimalField(max_digits=9, decimal_places=6)
+    city = models.CharField()
+    district = models.CharField()
+
+    minimumattendeecapacity = models.IntegerField(min_value=0)
```
Comment (berkatil) [inline back-end/sports_platform/sports_platform_api/models/event_models.py:33]: we can make the first letter of words capital to increase readability .( minimumAttendeeCapacity)

File: back-end/sports_platform/sports_platform_api/models/event_models.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        db_table = 'event_spectators'
+        unique_together = (('spectator', 'event'),)
+
+    spectator = models.ForeignKey('User', on_delete=models.CASCADE)
+    event = models.ForeignKey('Event', on_delete=models.CASCADE)
+
+
+class EventParticipationRequesters(models.Model):
+    class Meta:
+        db_table = 'event_spectators'
```
Comment (berkatil) [inline back-end/sports_platform/sports_platform_api/models/event_models.py:63]: table name should be changed.

File: back-end/sports_platform/sports_platform_api/models/event_models.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+from django.db import models
+
+
+creator: int
```
Comment (berkatil) [inline back-end/sports_platform/sports_platform_api/models/event_models.py:4]: I think we can delete these ones.

File: back-end/sports_platform/sports_platform_api/validation/event_validation.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+from rest_framework import serializers
+
+class Event(serializers.Serializer):
+
+    name = serializers.CharField(min_length=2, max_length=100, required=True)
+    sport = serializers.CharField(required=True)
+    description = models.CharField(required=False)
+
+    startDate = models.DateTimeField(format='iso-8601', required=True)
```
Comment (berkatil) [inline back-end/sports_platform/sports_platform_api/validation/event_validation.py:9]: I do not know if it does this but we should check that this should contain time and date together.

**Review by berkatil — CHANGES_REQUESTED** [review]
Thanks for your effort

File: back-end/sports_platform/sports_platform_api/validation/event_validation.py (hunk 4 above)
Comment (fileames) [reply back-end/sports_platform/sports_platform_api/validation/event_validation.py:9]: I could not find a validator specific to this. But I think checking if the string contains T would work. What do you think?

Comment (berkatil) [reply back-end/sports_platform/sports_platform_api/validation/event_validation.py:9]: sure, it makes sense, thanks!

**Pull request merged**